riff-raff/app/controllers/DeployController.scala

```diff
@@ -73,7 +73,10 @@ class DeployController(deployments: Deployments, prismLookup: PrismLookup, deplo
               "previewStacks" -> parameters.stacks.flatMap(_.nameOption).mkString(",")
             )
           case "deploy" =>
-            val uuid = deployments.deploy(parameters)
+            val uuid = deployments.deploy(parameters, requestSource = UserRequestSource(request.user)) match {
+              case Right(uuid) => uuid
+              case Left(error) => throw new IllegalStateException(error)
```

Comment:
Ideally we'd eliminate the throwing of exceptions entirely, but pulling the exception throwing up to the top level is a step in the right direction without making too many sweeping changes.

I've tweaked this to use `valueOr` from the cats either syntax.